packages/mdc-chips/chip/foundation.ts

```diff
@@ -189,15 +206,142 @@ export class MDCChipFoundation extends MDCFoundation<MDCChipAdapter> {
    * prevent the ripple from activating on interaction with the trailing icon.
    */
   handleTrailingIconInteraction(evt: MouseEvent | KeyboardEvent) {
-    const isEnter = (evt as KeyboardEvent).key === 'Enter' || (evt as KeyboardEvent).keyCode === 13;
-    evt.stopPropagation();
-    if (evt.type === 'click' || isEnter) {
+    if (this.shouldHandleInteraction_(evt)) {
       this.adapter_.notifyTrailingIconInteraction();
-      if (this.shouldRemoveOnTrailingIconClick_) {
-        this.beginExit();
-      }
+      this.removeChip_(evt);
+    }
+  }
+
+  /**
+   * Handles a keydown event from the root element.
+   */
+  handleKeydown(evt: KeyboardEvent) {
+    if (this.shouldRemoveChip_(evt)) {
+      return this.removeChip_(evt);
+    }
+
+    const key = evt.key;
+    // Early exit if the key is not usable
+    if (!navigationKeys.has(key)) {
+      return;
+    }
+
+    // Prevent default behavior for movement keys which could include scrolling
+    evt.preventDefault();
+    this.focusNextAction_(evt);
+  }
+
+  removeFocus() {
+    this.adapter_.setPrimaryActionAttr(strings.TAB_INDEX, '-1');
+    this.adapter_.setTrailingActionAttr(strings.TAB_INDEX, '-1');
+  }
+
+  focusAction(key: string, source: EventSource) {
```

Comment:
Since chip set already has key and source information, why do we not delegate this logic to chipset? Chip would just need to know where to focus (Primary or Trailing icon).

Reply:
That's a good question. We still need this logic somewhere, whether it's in the chip or the chip set. I'm flexible on where it goes. Let me know what you think.

Reply:
Params for this method seems confusing. It is accepting event information similar to `handle*` methods but it is a proxy to component API. I feel we need to move this logic to chip set and use Chip's low level APIs to focus the right element.

Something like:
- `chip.focusPrimary()`
- `chip.focusTrailingIcon()`


WDYT?

Reply:
That was my thinking too. Sounds good. I'll update with that.

Reply:
Done